src/core/maps/transferable_belief_model.h

```diff
@@ -0,0 +1,156 @@
+#ifndef SLAM_CTOR_CORE_TBM_H
+#define SLAM_CTOR_CORE_TBM_H
+
+#include <cassert>
+#include <vector>
+#include <string>
+
+/*------------------ declaration ---------------------*/
+
+//transferable belief model
+class TBM {
+public:
+  //public static belief types
+  static enum class Belief {
+    UNKNOWN,    //00
+    EMPTY,      //01
+    OCCUPIED,   //10
+    CONFLICT,   //11
+    MAX_BELIEF
+  } _belief_types;
+
+private:
+  //private variables
+  double _beliefs[static_cast<int>(Belief::MAX_BELIEF)];
+
+  //private methods
+  void set(Belief belief, double value);
+  void set_to_zero();
+  void set_to_default();
+
+public:
+  //public constructors
+  TBM() { set_to_default(); }
+  TBM(const TBM&) = default;
+  TBM& operator=(const TBM&) = default;
+  TBM& operator=(TBM&&) = default;
+  TBM(double unkown, double empty, double occupied, double conflict);
+	
+  //public methods
+  double get(Belief belief) const;
+  double unknown() const;
+  double empty() const;
+  double occupied() const;
+  double conflict() const;
+  void normalize();
+  void normalize_conflict();
+};
+
+TBM get_conjunctive(const TBM& lhs, const TBM& rhs);
+TBM get_disjunctive(const TBM& lhs, const TBM& rhs);
+
+/*------------------ definition ---------------------*/
+
+
+TBM::TBM(double unkown, double empty, double occupied, double conflict) {
+  set(TBM::Belief::UNKNOWN, unkown);
+  set(TBM::Belief::EMPTY, empty);
+  set(TBM::Belief::OCCUPIED, occupied);
+  set(TBM::Belief::CONFLICT, conflict);
+}
+
+void TBM::set(TBM::Belief belief, double value) {
+  assert(belief != TBM::Belief::MAX_BELIEF);
+  _beliefs[static_cast<int>(belief)] = value;
+}
+
+void TBM::set_to_zero() {
+  set(TBM::Belief::UNKNOWN, 0.0);
+  set(TBM::Belief::EMPTY, 0.0);
+  set(TBM::Belief::OCCUPIED, 0.0);
+  set(TBM::Belief::CONFLICT, 0.0);
+}
+
+void TBM::set_to_default() {
+  set(TBM::Belief::UNKNOWN, 1.0);
+  set(TBM::Belief::EMPTY, 0.0);
+  set(TBM::Belief::OCCUPIED, 0.0);
+  set(TBM::Belief::CONFLICT, 0.0);
+}
+
+double TBM::get(TBM::Belief belief) const {
+  assert(belief != TBM::Belief::MAX_BELIEF);
+  return _beliefs[static_cast<int>(belief)];
+}
+
+double TBM::unknown() const {
+  return get(TBM::Belief::UNKNOWN);
+}
+
+double TBM::empty() const {
+  return get(TBM::Belief::EMPTY);
+}
+
+double TBM::occupied() const {
+  return get(TBM::Belief::OCCUPIED);
+}
+
+double TBM::conflict() const {
+  return get(TBM::Belief::CONFLICT);
+}
+
+void TBM::normalize() {
+  double tot_weight {get(TBM::Belief::UNKNOWN) + get(TBM::Belief::EMPTY)
+    + get(TBM::Belief::OCCUPIED) + get(TBM::Belief::CONFLICT)};
```

Comment:
[Optional] Consider tabular alignment of the get calls. IMHO it makes code easier to read.
```
double tot_weight {get(UNKNOWN)  + get(EMPTY) +
                   get(OCCUPIED) + get(CONFLICT)};
```